src/approve.ts

```diff
@@ -52,7 +52,13 @@ export async function approve(
       if (
         review.user?.login == login &&
         review.commit_id == commit &&
-        review.state == "APPROVED"
+        review.state == "APPROVED" &&
+        ((
+          pull_request.data.requested_reviewers != undefined &&
+          pull_request.data.requested_reviewers.filter(reviewer => reviewer.login == login).length == 0
+        ) ||
+        pull_request.data.requested_reviewers == undefined
+        )
```

Comment:
This `if` statement is getting a little hard to follow 😅 

If we use the [optional chaining operator (`?.`)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) we can avoid both of the checks against `undefined` and remove the additional `||`, which simplifies it a bit.

Additionally, [`!Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is a slightly more concise way of expressing `filter(...).length == 0`.

```suggestion
        !pull_request.data.requested_reviewers?.some(
          (user) => user.login === login
        )
```

Reply:
Hi @hmarr, corrected the format and used your suggestions, can you help to check? 😊